src/app/services/reducers/query-runner-reducers.ts

```diff
@@ -1,17 +1,19 @@
 import { IAction } from '../../../types/action';
-import { CLEAR_RESPONSE, QUERY_GRAPH_SUCCESS,
-  SET_SAMPLE_QUERY_SUCCESS, VIEW_HISTORY_ITEM_SUCCESS } from '../redux-constants';
+import {
+  CLEAR_RESPONSE, QUERY_GRAPH_RUNNING,
+  QUERY_GRAPH_SUCCESS, SET_SAMPLE_QUERY_SUCCESS, VIEW_HISTORY_ITEM_SUCCESS
+} from '../redux-constants';
 
 export function graphResponse(state = {}, action: IAction): any {
   switch (action.type) {
     case QUERY_GRAPH_SUCCESS:
       return action.response;
     case VIEW_HISTORY_ITEM_SUCCESS:
       return action.response;
-    case SET_SAMPLE_QUERY_SUCCESS:
+    case QUERY_GRAPH_RUNNING:
```

Comment:
Please elaborate on why we made this change. I have not been able to figure out from going through the rest of the code.

Reply:
Once we are running a new query it removes the previous results.

I removed the SET_SAMPLE_QUERY_SUCCESS listener because it was the main cause of the bug

Reply:
Why did we replace SET_SAMPLE_QUERY_SUCCESS with QUERY_GRAPH_RUNNING?

Reply:
The replacement is because as a query is running, we do not need to see the response of the previously ran request.

Reply:
Notice that it is also at this action `QUERY_GRAPH_RUNNING` that we clear the response status

Reply:
For this use case. we'll be fine having neither `SET_SAMPLE_QUERY_SUCCESS` nor `QUERY_GRAPH_RUNNING` cases. So we can delete the `QUERY_GRAPH_RUNNING` case here.